docs/data/data-grid/column-dimensions/ColumnAutosizingGroupedRows.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import {
+  DataGridPremium,
+  useGridApiRef,
+  useKeepGroupedColumnsHidden,
+} from '@mui/x-data-grid-premium';
+import { useMovieData } from '@mui/x-data-grid-generator';
+
+export default function ColumnAutosizingGroupedRows() {
+  const data = useMovieData();
+  const apiRef = useGridApiRef();
+
+  const columns = React.useMemo(() => {
+    return data.columns.map((col) => ({ ...col, width: 50 }));
+  }, [data.columns]);
+
+  const initialState = useKeepGroupedColumnsHidden({
+    apiRef,
+    initialState: { rowGrouping: { model: ['company'] } },
+  });
+
+  React.useEffect(() => {
+    apiRef.current.subscribeEvent('rowExpansionChange', (params) => {
```

Comment:
```suggestion
    return apiRef.current.subscribeEvent('rowExpansionChange', (params) => {
```

I forgot to return a cleanup function in my CS demo 😅